app-backend/db/src/main/scala/SceneToLayerDao.scala

```diff
@@ -39,54 +46,60 @@ object SceneToLayerDao
     FROM
   """ ++ tableF
 
+  def mosaicDefCacheKey(projectLayerId: UUID): String =
+    s"SceneToLayer:ProjectLayer:$projectLayerId:MultiTiff:${Config.publicData.enableMultiTiff}"
+
+  def deleteMosaicDefCache(projectLayerId: UUID): ConnectionIO[Unit] = {
+    for {
+      _ <- {
+        val cacheKey = mosaicDefCacheKey(projectLayerId)
+        logger.debug(s"Removing $cacheKey")
+        remove(cacheKey)(mosaicDefinitionCache, async[ConnectionIO]).attempt
+      }
+    } yield ()
+  }
+
   def acceptScene(projectLayerId: UUID, sceneId: UUID): ConnectionIO[Int] = {
-    fr"""
+    for {
+      acceptCount <- fr"""
       UPDATE scenes_to_layers
       SET accepted = true
       WHERE project_layer_id = ${projectLayerId}
         AND scene_id = ${sceneId}
-    """.update.run
+        """.update.run
+      _ <- deleteMosaicDefCache(projectLayerId)
+    } yield acceptCount
   }
 
   def acceptScenes(
       projectLayerId: UUID,
       sceneIds: List[UUID]
   ): ConnectionIO[Int] = {
-    sceneIds.toNel match {
-      case Some(ids) => acceptScenes(projectLayerId, ids)
-      case _         => 0.pure[ConnectionIO]
-    }
+    for {
+      updateCount <- sceneIds.toNel match {
+        case Some(ids) => acceptScenes(projectLayerId, ids)
+        case _         => 0.pure[ConnectionIO]
+      }
+      _ <- deleteMosaicDefCache(projectLayerId)
+    } yield updateCount
   }
 
   def acceptScenes(
       projectLayerId: UUID,
       sceneIds: NEL[UUID]
   ): ConnectionIO[Int] = {
-    (
-      fr"""
+    for {
+      updateCount <- (
+        fr"""
         UPDATE scenes_to_layers
         SET accepted = true
       """ ++ Fragments.whereAnd(
-        fr"project_layer_id = ${projectLayerId}",
-        Fragments.in(fr"scene_id", sceneIds)
-      )
-    ).update.run
-  }
-
-  def addSceneOrdering(projectId: UUID): ConnectionIO[Int] = {
-    (fr"""
-      UPDATE scenes_to_layers
-      SET scene_order = s.rnum
-      FROM (
-        SELECT scene_id, project_layer_id, row_number() over (ORDER BY stl.scene_order ASC, scenes.acquisition_date ASC, scenes.cloud_cover ASC) as rnum
-        FROM scenes_to_layers stl
-          JOIN scenes ON scenes.id = stl.scene_id
-          JOIN project_layers pl ON pl.id = stl.project_layer_id
-          WHERE pl.project_id = ${projectId}
-      ) s
-      WHERE scenes_to_layers.project_layer_id = s.project_layer_id
-        AND scenes_to_layers.scene_id = s.scene_id
-    """).update.run
+          fr"project_layer_id = ${projectLayerId}",
+          Fragments.in(fr"scene_id", sceneIds)
+        )
+      ).update.run
+      _ <- deleteMosaicDefCache(projectLayerId)
```

Comment:
same as above

Reply:
I think this case is different from above because `sceneIds` are a non empty list so there will be at least one updated